bundles/org.openhab.binding.freeboxos/src/main/java/org/openhab/binding/freeboxos/internal/api/rest/FreeboxOsSession.java

```diff
@@ -95,9 +99,13 @@ public void initialize(FreeboxOsConfiguration config) throws FreeboxException, I
 
     public void openSession(String appToken) throws FreeboxException {
         Session newSession = getManager(LoginManager.class).openSession(appToken);
-        getManager(WebSocketManager.class).openSession(newSession.sessionToken(), wsReconnectInterval);
         session = newSession;
         this.appToken = appToken;
+        if (vmSupported == null) {
+            vmSupported = getManager(SystemManager.class).getConfig().modelInfo().hasVm();
+        }
+        getManager(WebSocketManager.class).openSession(newSession.sessionToken(), wsReconnectInterval,
+                vmSupported == null ? false : vmSupported.booleanValue());
```

Comment:
This introduces a compiler warning:
```
[WARNING] /var/jenkins_home/workspace/PR-openHAB-Addons/bundles/org.openhab.binding.freeboxos/src/main/java/org/openhab/binding/freeboxos/internal/api/rest/FreeboxOsSession.java:[108,47] Potential null pointer access: this expression has a '@Nullable' type
```

Also consider:
```java
Boolean.TRUE.equals(vmSupported)
```

which also handles the null case.

Reply:
Good catch, thank you.